### ISSUE TITLE
Fix URI encoding of forward slashes in Service Worker mode

### ISSUE THREAD

**Issue by Jaifroid** [kiwix/kiwix-js#506]
This is the underlying issue corresponding to PR #494. Here is a summary of the issue:

There is a problem in the way we currently set the iframe URL for the Service Worker The problem is [this line](https://github.com/kiwix/kiwix-js/blob/9cc7cb9ff04576e33cd69420856c1afb6af7a472/www/js/app.js#L776):
```
iframeArticleContent.src = dirEntry.namespace + "/" + encodeURIComponent(dirEntry.url);
```
This sets the source of the iframe to the encoded form of the URL, but `encodeURIComponent` encodes forward slashes, so a url like `A/Singapore/Riverside` will be set as `A/Singapore%2FRiverside`. While the file will be read correctly by the browser, the `%2F` causes the SW to think it's in the wrong directory as far as any relative asset links on the page are concerned.

The reason we use `encodeURIComponent` at this point is to solve another problem, as per [discussion here](https://github.com/kiwix/kiwix-js/issues/428#issuecomment-429742993). It was a rather specialized issue.

Additionally, we set the relative location of the document loaded in the iframe incorrectly. It should be:
```
iframeArticleContent.src = '../' + dirEntry.namespace + "/" + dirEntry.url;
```

We still need to encodeURI any elements of the URL that are not a forward slash so that footnotes of type `href="Ottoman_Empire.html#cite_note-:0-13"` still work, as well as footnotes in articles such as "Presidential $1 Coin Program".

**Comment by mossroy** [comment]
#494 is not a PR : I suppose you're talking about #498?

Could you give steps to reproduce the issue?

**Comment by mossroy** [comment]
Fixed by #498
